docs/kb/semgrep-code/scan-engine-kill.md

```diff
@@ -0,0 +1,146 @@
+            recommend using 1 job if you are running out of memory.
+          3. Scan the repo in parts (contact us for help)
+
+          Otherwise, it is likely that semgrep is hitting the limit on only some
+          files. In this case, you can try to set the limit on the amount of memory
+          semgrep can use on each file with `--max-memory <memory>`. We recommend
+          lowering this to a limit 70% of the available memory. For CI runs with
+          interfile analysis, the default max-memory is 5000MB. Without, the default
+          is unlimited.
+
+          The last thing you can try if none of these work is to raise the stack
+          limit with `ulimit -s <limit>`.
+
+          If you have tried all these steps and still are seeing this error, please
+          contact us.
+
+           Error: semgrep-core exited with unexpected output
+```
+
+## Determining the size of your monorepository 
+
+By default, Semgrep places resource limitations on the size of file scanned and memory allocated.
+
+However, Semgrep does not place limitations on the number of files scanned and scanning a large monorepository can involve thousands of lines of code. Thus, the first course of action is to determine exactly how many files are getting scanned:
+
+1. View your CI logs. This step depends on your CI provider.
+2. Search for the section "Scan Status".
+
+The following is an example of a large monorepository:
+
+
+```console
+┌─────────────┐
+│ Scan Status │ 
+└─────────────┘ 
+  Scanning 91749 files tracked by git with 1068 Code rules, 498 Pro rules:
+            
+  Language      Rules    Files          Origin      Rules
+ ──────────────────────────────        ───────────────────
+  <multilang>      60   199251          Community     551
+  ts              166    26672          Pro rules     498
+  python          314     8089          Custom         19
+  scala            13     5415
+  json              1     4149       
+  yaml              7     1952
+  js              160     1084      
+  terraform        13      470      
+  bash              4      408           
+  go               95      228
+  ruby             22      228
+  php              25       99
+  swift            41       89
+  html              1       72         
+  dockerfile        2       31           
+  c                 9       16
+  rust             49        8
+  java            156        2
+  kotlin           47        1
+```
+
+Now you have a good idea of the size of your monorepository. After establishing the size and breakdown of programming languages and files, you can decide what adjustments to take for a scan to succeed.
+
+## Scanning components separately 
+
+Based on the composition provided by the logs, you may be able to determine if your repository is modular. If so, you can try scanning the components separately.
+
+Interfile analysis still applies in the smaller scope assuming the modules are disparate. 
```

Comment:
maybe a link to a definition of interfile analysis here?

Reply:
Ye! Rather bare of links, this article is.